forest/cli/src/cli/snapshot_cmd.rs

```diff
@@ -335,3 +491,25 @@ fn delete_snapshot(snapshot_path: &PathBuf) {
         }
     }
 }
+
+#[cfg(feature = "rocksdb")]
+type Database = forest_db::rocks::RocksDb;
+
+#[cfg(feature = "paritydb")]
+type Database = forest_db::parity_db::ParityDb;
+
+fn open_db(
+    path: &std::path::Path,
+    #[cfg(feature = "rocksdb")] config: &Config,
+) -> Result<Database, anyhow::Error> {
+    #[cfg(feature = "rocksdb")]
+    {
+        forest_db::rocks::RocksDb::open(path, &config.rocks_db)
+            .map_err(|e| anyhow::anyhow!("failed to open db: {}", e))
+    }
+    #[cfg(feature = "paritydb")]
+    {
+        let paritydb_config = forest_db::parity_db::ParityDbConfig::from_path(path);
+        forest_db::parity_db::ParityDb::open(&paritydb_config)
+    }
+}
```

Comment:
This got a bit complex; too many compilation conditionals lurking around, even as a param - supporting another config would be a pain. How about we create two `open_db` methods, i.e.

```
#[cfg(feature = "rocksdb")]
fn open_db(someparams) { ... }
#[cfg(feature = "paritydb")]
fn open_db(otherparams) { ... }
```